pcdsdevices/pseudopos.py

```diff
@@ -0,0 +1,83 @@
+from ophyd.device import Component as Cpt
+from ophyd.pseudopos import PseudoPositioner, PseudoSingle
+
+
+class SyncAxesBase(PseudoPositioner):
+    """
+    Synchronized Axes.
+
+    This will move all axes in a coordinated way, retaining offsets.
+
+    This can be configured to report its position as the min, max, mean, or any
+    custom function acting on a list of positions. Min is the default.
+
+    You should subclass this by adding real motors as components. The class
+    will pick them up and include them correctly into the coordinated move.
+
+    Parameters
+    ----------
+    position_mode: ``func``, optional
+        The function to apply to the list of current positions to determine
+        the combined position. By default, this is the minimum of all axis
+        positions.
+
+    Attributes
+    ----------
+    axes: ``list of str``
+        Names of the components that correspond to the real axes.
+    """
+    pseudo = Cpt(PseudoSingle)
+
+    _default_mode = min
+
+    def __init__(self, *args, position_mode=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._mode = position_mode or self._default_mode
+        self._offsets = {}
+
+    def save_offsets(self, **offsets):
+        """
+        Save the current offsets for the synchronized assembly.
+
+        If not done earlier, this will be automatically run before it is first
+        needed (generally, right before the first move).
+
+        Note that this method does not protect you from getting your offsets
+        into an inconsistent state! Pass arguments into the `offsets` param at
+        your own risk.
+
+        Parameters
+        ----------
+        **offsets: ``int`` or ``float``, optional
```

Comment:
Tempted to say this is too much power. Maybe they just adjust the motors and `save_offsets`...

Reply:
I'm tempted to agree, it's not hard to remove